web_tool_script_2.R

```diff
@@ -197,3 +197,6 @@ if (file.exists(bonds_inputs_file)){
   }
 }
 
+invisible(set_portfolio_parameters(file_path = fs::path(par_file_path,paste0(portfolio_name_ref_all,"_PortfolioParameters.yml"))))
```

Comment:
I'm not sure if this is a correct use of `invisible()`. @jacobvjk, can you explain once again what you intend to do?

Reply:
My intention was simply to make the parameters available as objects again when running script 2. This was the case when running the scripts interactively, probably because they stay in the global environment, but not when they are called from the command line one after the other. The stress test script needs these parameters however. But maybe there i no point in running this invisibly..

Reply:
Thanks. Yeah, I think `invisible()` achieves nothing. But if your experiment suggests that you do, then prefer the evidence of the experiment than my understanding from only running the code in my head.

For the record, this is what we are talking about:

```r
set_portfolio_parameters <- function(file_path){

  cfg <- config::get(file = file_path)

  portfolio_name_ref <<- cfg$parameters$portfolio_name_ref
  portfolio_name_in <<- cfg$parameters$portfolio_name_in
  investor_name_in <<- cfg$parameters$investor_name_in

}
```

Note that  using `<<-` in a function is generally a bad idea, unless you are doing something [like this](http://adv-r.had.co.nz/Functional-programming.html):

<img src=https://user-images.githubusercontent.com/5856545/95363975-d3b5fa00-0895-11eb-8d98-c7c8c6ffac42.png width=600>


If these objects need to be assigned to the global environment, then it's simpler and clearer to do just that:

```r
  cfg <- config::get(file = file_path)
  portfolio_name_ref <- cfg$parameters$portfolio_name_ref
  portfolio_name_in <- cfg$parameters$portfolio_name_in
  investor_name_in <- cfg$parameters$investor_name_in
```

And if the function is only a way to reuse this code snippet, then you can
store the code snippet in a dry/file.R and source it wherever you need it.

Reply:
This is actually a very common practice in all versions of the PACTA_analysis repo, so glad to pick this up and change it across the repo soon. Maybe once we merge into 'Master', this would be a good idea